### ISSUE TITLE
Address compilation warnings on Windows GCC7.1

### ISSUE THREAD

**Issue by Lestropie** [MRtrix3/mrtrix3#1247]
Example:

```
In file included from ./core/mrtrix.h:36:0,
                 from ./core/cmdline_option.h:27,
                 from ./core/app.h:27,
                 from ./core/command.h:22,
                 from cmd/mrthreshold.cpp:18:
./core/types.h:284:40: warning: optimization attribute on 'uint64_t std::abs(uint64_t)' follows definition but the attribute doesn't match [-Wattributes]
   FORCE_INLINE uint64_t abs (uint64_t x) { return x; }
                                        ^
In file included from C:/msys64/mingw64/include/c++/7.2.0/cstdlib:77:0,
                 from C:/msys64/mingw64/include/c++/7.2.0/ext/string_conversions.h:41,
                 from C:/msys64/mingw64/include/c++/7.2.0/bits/basic_string.h:6159,
                 from C:/msys64/mingw64/include/c++/7.2.0/string:52,
                 from C:/msys64/mingw64/include/c++/7.2.0/stdexcept:39,
                 from C:/msys64/mingw64/include/c++/7.2.0/array:39,
                 from C:/msys64/mingw64/include/c++/7.2.0/tuple:39,
                 from C:/msys64/mingw64/include/c++/7.2.0/bits/stl_map.h:63,
                 from C:/msys64/mingw64/include/c++/7.2.0/map:61,
                 from cmd/mrthreshold.cpp:16:
C:/msys64/mingw64/include/c++/7.2.0/bits/std_abs.h:61:3: note: previous definition of 'long long int std::abs(long long int)' was here
   abs(long long __x) { return __builtin_llabs (__x); }
```

Tried to address in 2a508473, but reverted in f4a96a85.

Need to experiment again to try to find appropriate pre-compiler conditions to use in `core/types.h` that will work silently with all compilers / OS's.

**Comment by Lestropie** [comment]
Boosting this to `3.0_RC3` milestone. Exceptionally frustrating, and hasn't gone away as of `g++` 7.3.0.

**Comment by jdtournier** [comment]
closed with #1304
